v0.9.0 panics on OpenBSD

<!-- Please report security issues by email to security@ipfs.io -->

#### Version information:

go-ipfs version: 0.9.0
Repo version: 11
System version: amd64/openbsd
Golang version: go1.16.5
<!-- Output From `ipfs version --all`

Please check dist.ipfs.io for a newer version of go-ipfs and update if necessary. Report back if the problem persists.

If you can't run `ipfs version --all` or that command fails, include as much information as you can: IPFS version, computer architecture (e.g., Intel x86 64bit), operating system, etc. -->

#### Description:
<!-- This is where you get to tell us what went wrong. When doing so, please make sure to include *all* relevant information.

Please try to include:
  * What you were doing when you experienced the bug.
  * Any error messages you saw, *where* you saw them, and what you believe may have caused them (if you have any ideas).
  * When possible, steps to reliably produce the bug.
-->
After running for a period of time, `ipfs daemon` panics with the following:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x9f7872]

goroutine 270 [running]:
github.com/multiformats/go-multiaddr/net.(*maListener).Accept(0xc0006b49a0, 0xc00287de18, 0xb23a5d, 0xc000019980, 0xc008136a40)
        /build/pobj/go-ipfs-0.9.0/go/pkg/mod/github.com/multiformats/go-multiaddr@v0.3.2/net/net.go:259 +0x112
github.com/libp2p/go-tcp-transport.(*tcpListener).Accept(0xc0007af950, 0xc00287de70, 0xc00287de78, 0x20, 0xc0007d8a80)
        /build/pobj/go-ipfs-0.9.0/go/pkg/mod/github.com/libp2p/go-tcp-transport@v0.2.2/tcp.go:66 +0x37
github.com/libp2p/go-tcp-transport.(*tracingListener).Accept(0xc0001636f0, 0x0, 0x0, 0xc001438fc0, 0x1adbf60)
        /build/pobj/go-ipfs-0.9.0/go/pkg/mod/github.com/libp2p/go-tcp-transport@v0.2.2/metrics.go:226 +0x37
github.com/libp2p/go-libp2p-transport-upgrader.(*listener).handleIncoming(0xc001438fc0)
        /build/pobj/go-ipfs-0.9.0/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.4.2/listener.go:77 +0xf7
created by github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).UpgradeListener
        /build/pobj/go-ipfs-0.9.0/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.4.2/upgrader.go:49 +0x18d
```

The daemon also produces a plethora of this error while running:
```
2021-06-23T06:01:16.574-0600    ERROR   tcp-tpt go-tcp-transport@v0.2.2/tcp.go:45       Failed set keepalive period: set tcp4 REDACTED:1777->REDACTED:30504: protocol not available
```

--- a/go.mod
+++ b/go.mod
@@ -60,7 +60,7 @@ require (
 	github.com/jbenet/go-temp-err-catcher v0.1.0
 	github.com/jbenet/goprocess v0.1.4
 	github.com/libp2p/go-doh-resolver v0.3.1
-	github.com/libp2p/go-libp2p v0.14.2
+	github.com/libp2p/go-libp2p v0.14.3
 	github.com/libp2p/go-libp2p-circuit v0.4.0
 	github.com/libp2p/go-libp2p-connmgr v0.2.4
 	github.com/libp2p/go-libp2p-core v0.8.5
@@ -82,12 +82,12 @@ require (
 	github.com/libp2p/go-libp2p-tls v0.1.3
 	github.com/libp2p/go-libp2p-yamux v0.5.4
 	github.com/libp2p/go-socket-activation v0.0.2
-	github.com/libp2p/go-tcp-transport v0.2.2
+	github.com/libp2p/go-tcp-transport v0.2.3
 	github.com/libp2p/go-ws-transport v0.4.0
 	github.com/lucas-clemente/quic-go v0.21.1
 	github.com/miekg/dns v1.1.41
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/multiformats/go-multiaddr v0.3.2
+	github.com/multiformats/go-multiaddr v0.3.3
 	github.com/multiformats/go-multiaddr-dns v0.3.1
 	github.com/multiformats/go-multibase v0.0.3
 	github.com/multiformats/go-multihash v0.0.15

